dali/operators/decoder/audio/audio_decoder_op.h

```diff
@@ -0,0 +1,134 @@
+    auto &input = ws.template InputRef<Backend>(0);
+    const auto batch_size = input.shape().num_samples();
+
+    for (int i = 0; i < input.shape().num_samples(); i++) {
+      DALI_ENFORCE(input.shape()[i].size() == 1, "Input must be 1D encoded byte data");
+    }
+    DALI_ENFORCE(IsType<uint8_t>(input.type()), "Input must be stored as uint8_t data.");
+    TypeInfo type;
+    TypeInfo type_i32;
+    type_i32.SetType<int>(DALI_INT32);
+    decoders_.resize(batch_size);
+
+    TYPE_SWITCH(output_type_, type2id, OutputType, (int16_t, int32_t, float), (
+            for (int i=0; i < batch_size; i++) {
+            decoders_[i] = std::make_unique<GenericAudioDecoder<OutputType>>();
+        }
+        type.SetType<OutputType>(output_type_);
+    ), DALI_FAIL("Unsupported output type"))  // NOLINT
+
+    output_desc.resize(detail::kNumOutputs);
+
+    // Currently, metadata is only the sampling rate.
+    // On the event something else would emerge,
+    // this approach should be completely redefined
+    TensorListShape<> shape_rate(batch_size, 1);
+    TensorListShape<> shape_data(batch_size, 2);
+
+    for (int i = 0; i < batch_size; i++) {
+      auto meta = decoders_[i]->Open({reinterpret_cast<const char *>(input[i].raw_mutable_data()),
+                                      input[i].shape().num_elements()});
+      samples_meta_.emplace_back(meta);
+      shape_data.set_tensor_shape(i, {meta.length, meta.channels});
+      shape_rate.set_tensor_shape(i, {1});
+      files_names_.emplace_back(input[i].GetSourceInfo());
+    }
+    output_desc[0] = {shape_data, type};
+    output_desc[1] = {shape_rate, type_i32};
+    return true;
+  }
+
+
+  void RunImpl(workspace_t<Backend> &ws) override {
+    auto &decoded_output = ws.template OutputRef<Backend>(0);
+    auto &sample_rate_output = ws.template OutputRef<Backend>(1);
+    auto &tp = ws.GetThreadPool();
+
+    for (int i = 0; i < decoded_output.shape().num_samples(); i++) {
+      tp.DoWorkWithID([&, i](int thread_id) {
+          try {
+            decoders_[i]->Decode({reinterpret_cast<char *>(decoded_output[i].raw_mutable_data()),
+                                  static_cast<int>(decoded_output[i].type().size() *
+                                                   decoded_output[i].shape().num_elements())});
+            auto sample_rate_ptr =
+                    reinterpret_cast<sample_rate_t *>(sample_rate_output[i].raw_mutable_data());
+            *sample_rate_ptr = samples_meta_[i].sample_rate;
+          } catch (const DALIException &e) {
+            DALI_FAIL(make_string("Error decoding file.\nError: ", e.what(),
+                                  "\nFile: ", files_names_[i], "\n"));
+          }
+      });
+    }
+  }
+
+
+  bool CanInferOutputs() const override {
+    return true;
+  }
+
+
+ private:
+  DALIDataType output_type_;
+  std::vector<std::string> files_names_;
+  std::vector<AudioMetadata> samples_meta_;
```

Comment:
Grammar Nazi: overuse of plural.
```suggestion
  std::vector<AudioMetadata> sample_meta_;
```

Reply:
Doned ;)